scene/resources/particles_material.cpp

```diff
@@ -316,14 +316,17 @@ void ParticlesMaterial::_update_shader() {
 
 	if (flags[FLAG_DISABLE_Z]) {
 
-		code += "		float angle1_rad = atan(direction.y, direction.x) + rand_from_seed_m1_p1(alt_seed) * spread_rad;\n";
+		code += "		float angle1_rad = rand_from_seed_m1_p1(alt_seed) * spread_rad;\n";
+		code += "		angle1_rad += direction.x != 0.0 ? atan(direction.y, direction.x) : sign(direction.y) * (pi / 2.0);\n";
```

Comment:
This makes the previous statement unused.

Reply:
Nevermind, I misread.

Reply:
Do you mean in the conditional, so `atan(direction.y, direction.x)`? If you set a direction x=0.5, y=1 (z is ignored because disable z flag) for example, it gets used.

Reply:
I missed that the line used `+=` operator, and though it was `=`, thus replacing the previous definition.